src/molusce/moluscedialog.py

```diff
@@ -379,6 +387,65 @@ def addFactor(self):
         self.__updateAnalyticTabs(self.geometry_matched)
         gc.collect()
 
+    def addFactorSeparateVars(self):
+        layerNames = self.lstLayersSeparateVars.selectedItems()
+
+        if len(layerNames) <= 0:
+            QMessageBox.warning(
+                self,
+                self.tr("Missed selected row"),
+                self.tr(
+                    "Factor raster is not selected. Please specify input data and try again"
+                ),
+            )
+            return
+
+        for i in layerNames:
+            layerName = i.text()
+
+            if len(self.lstFactorsSeparateVars.findItems(layerName, Qt.MatchExactly)) > 0:
+                return
+
+            item = QListWidgetItem(i)
+            layerId = str(item.data(Qt.UserRole))
+            self.lstFactorsSeparateVars.insertItem(self.lstFactors.count() + 1, item)
+
+            try:
+                if "factors_sim" in self.inputs:
+                    self.inputs["factors_sim"][layerId] = Raster(
+                        str(utils.getLayerById(layerId).source()),
+                        utils.getLayerMaskById(layerId),
+                    )
+                else:
+                    d = dict()
+                    d[layerId] = Raster(
+                        str(utils.getLayerById(layerId).source()),
+                        utils.getLayerMaskById(layerId),
+                    )
+                    self.inputs["factors_sim"] = d
+
+                self.inputs["bandCount_sim"] = self.__bandCount(sim=True)
+
+                self.logMessage(self.tr("Added factor (sim) layer %s") % (layerName))
+            except MemoryError:
+                self.logErrorReport(
+                    self.tr(
+                        "Memory Error occurred (loading raster %s). Perhaps the system is low on memory."
+                    )
+                    % (layerName)
+                )
+                QMessageBox.warning(
+                    self,
+                    self.tr("Memory error"),
+                    self.tr(
+                        "Memory error occurred. Perhaps the system is low on memory."
+                    ),
+                )
+                raise
+                return
+        self.consistency_sim_checked = False
+        gc.collect()
```

Comment:
There seems to be no point in using the `gc.collect `without a specific reason here and elsewhere in the PR